src/lib/peers/PeerManager.ts

```diff
@@ -38,40 +35,42 @@ const keybaseDiscovery: SocialDiscovery = {
 };
 
 type PeerManagerMetadata = {
-  localPeerInfo: PeerInfo | null;
+  peerInfo: PeerInfo | null;
 };
 
 class PeerManager {
   private fileSystem: typeof fs;
 
-  private metadataPath: string;
-  private metadata: PeerManagerMetadata = { localPeerInfo: null };
+  private peerInfoMetadataPath: string;
+  private peerStoreMetadataPath: string;
 
-  private localPeerInfo: PeerInfo;
+  peerInfo: PeerInfo;
   private peerStore: Map<string, PeerInfo>;
   private keyManager: KeyManager;
   multicastBroadcaster: MulticastBroadcaster;
-  private socialDiscoveryServices: SocialDiscovery[];
+  socialDiscoveryServices: SocialDiscovery[];
 
   // Peer connections
-  server: grpc.Server;
-  serverStarted: boolean = false;
-  private gitBackend: GitBackend;
-  private credentials: grpc.ServerCredentials;
-  private peerConnections: Map<string, GitFrontend>;
+  peerServer: PeerServer;
+  private peerConnections: Map<string, PeerConnection>;
+  turnClient: TurnClient;
+
+  private stealthMode: boolean;
 
   constructor(
```

Comment:
To me the peer manager is only meant to manage the life cycle of peers and mapping of state to peers.

All the other peer stuff should be done by subobjects. You can have something dealing with nat/turn... etc. Then add them as dependencies to PeerManager via dependency injection. Or even do object composition, right now the PeerManager has a lot of functionality.

Reply:
Especially any social discovery should be outside of this module entirely. It's a totally different abstraction level of functionality.